### PR TITLE
fix: add adminRoles to stash

### PR DESCRIPTION
**Pull request by dpilch**
#### Description of changes

Fix regression from https://github.com/aws-amplify/amplify-category-api/pull/1919. Add admin roles to stash.

#### Issue #, if available

https://github.com/aws-amplify/amplify-category-api/issues/2060

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-function-transformer/src/graphql-function-transformer.ts [hunk 1]
```diff
@@ -145,6 +145,13 @@ export class FunctionTransformer extends TransformerPluginBase {
             qref(`$ctx.stash.put("authRole", "arn:aws:sts::${account}:assumed-role/${authRole}/CognitoIdentityCredentials")`),
             qref(`$ctx.stash.put("unauthRole", "arn:aws:sts::${account}:assumed-role/${unauthRole}/CognitoIdentityCredentials")`),
           );
+
+          const identityPoolId = context.synthParameters.identityPoolId;
+          if (identityPoolId) {
+            requestTemplate.push(qref(`$util.qr($ctx.stash.put("identityPoolId", "${identityPoolId}"))`));
```
Comment (alharris-at) [inline packages/amplify-graphql-function-transformer/src/graphql-function-transformer.ts:151]: You probably don't want the `$util.qr` here I'd imagine.

**Comment by alharris-at** [comment]
I think we need at a minimum some sort of test which is going to demonstrate the change to the generated resolver.

File: packages/amplify-graphql-function-transformer/src/__tests__/amplify-graphql-function-transformer.test.ts [hunk 2]
```diff
@@ -345,3 +346,26 @@ test('@function directive applied to Object should throw Error', () => {
     }),
   ).toThrow('Directive "@function" may not be used on OBJECT.');
 });
+
+test.only('includes auth info in stash', () => {
```
Comment (AaronZyLee) [inline packages/amplify-graphql-function-transformer/src/__tests__/amplify-graphql-function-transformer.test.ts:350]:
```suggestion
test('includes auth info in stash', () => {
```

**Review by AaronZyLee — COMMENTED** [review]
Overall LGTM. A minor change below.

**Review by alharris-at — COMMENTED** [review]
LGTM - will shipit once the change that Aaron suggested is merged in.

**Review by alharris-at — APPROVED** [review]

**Review by AaronZyLee — APPROVED** [review]

**Comment by DevTGhosh** [comment]
Any idea when this will be part of the amplify cli release? It's breaking a major functionality for us. It's still an issue as of cli version 12.8.2

**Pull request merged**